src/main/scala/org/broadinstitute/hail/sparkextras/OrderedRDD.scala

```diff
@@ -290,12 +330,61 @@ class OrderedRDD[T, K, V] private(rdd: RDD[(K, V)],
     assert(newPartEnd.last == n - 1)
     assert(newPartEnd.zip(newPartEnd.tail).forall { case (i, inext) => i <= inext })
 
-    val newRangeBounds = newPartEnd.init.map(end =>
-      orderedPartitioner.rangeBounds(end))
+    val newRangeBounds = newPartEnd.init.map(orderedPartitioner.rangeBounds)
+
+    new OrderedRDD[PK, K, V](new BlockedRDD(persisted, newPartEnd), OrderedPartitioner(newRangeBounds))
+  }
+}
+
+trait OrderedKey[PK, K] extends Serializable {
+  /**
+    * This method must be monotonic in {@code k}
+    */
+  def project(key: K): PK
+
+  implicit def kOrd: Ordering[K]
+
+  implicit def pkOrd: Ordering[PK]
+
+  implicit def kct: ClassTag[K]
+
+  implicit def pkct: ClassTag[PK]
```

Comment:
To make exposing the implicits a little more concise, we could write this as:

```
trait OrderedKey[PK, K] extends Serializable { self =>
  def project(key: K): PK
  def kOrd: Ordering[K]
  ...
  object implicits {
    implicit val kOrd = kOrd
    ...
  }
}
```

Then, when we want to pull out the implicits, we can just write `import kOk.implicits._`.